src/stacked_borrows/diagnostics.rs

```diff
@@ -447,14 +452,16 @@ impl<'span, 'history, 'ecx, 'mir, 'tcx> DiagnosticCx<'span, 'history, 'ecx, 'mir
     }
 
     #[inline(never)] // This is only called on fatal code paths
-    pub fn dealloc_error(&self) -> InterpError<'tcx> {
+    pub fn dealloc_error(&self, stack: &Stack) -> InterpError<'tcx> {
         let Operation::Dealloc(op) = &self.operation else {
             unreachable!("dealloc_error should only be called during a deallocation")
         };
         err_sb_ub(
             format!(
-                "no item granting write access for deallocation to tag {:?} at {:?} found in borrow stack",
-                op.tag, self.history.id,
+                "attemtping deallocation using {tag:?} at {alloc_id:?}{cause}",
```

Comment:
There is a small inconsistency here in the error message, when compared with accesses... accesses say `attempting a read/write access`, but here we leave away the `a`. IMO that sounds better. In fact `attempting read/write access` also works, doesn't it? Should we switch to that?

Reply:
I think this is that this is an inconsistency in English, at least as it is spoken. I follow your logic, but the words sound wrong to me.

Reply:
:shrug: